src/nvme/tree.c

```diff
@@ -1156,7 +1156,7 @@ int nvme_init_ctrl(nvme_host_t h, nvme_ctrl_t c, int instance)
 	if (strcmp(c->transport, "loop")) {
 		c->address = nvme_get_attr(path, "address");
 		if (!c->address) {
-			errno = ENXIO;
+			errno = ENVME_CONNECT_INVAL_TR;
```

Comment:
All the other functions called from nvmf_add_ctrl() would return the error code directly; nvme_init_ctrl() is the only one returning -1 and setting errno.
We need to be consistent here; suggestion is to have all functions from nvmf_add_ctrl() return -1 and setting the errno.

Reply:
Yes, makes sense.